### PR TITLE
legacy: extract LegacyArchives into legacy/archives.py, add ArchivesInterface, refs #9556

### PR DESCRIPTION
**Pull request by mr-raj12**
Moves the Borg 1.x archive list management out of `Archives` and into `LegacyArchives`, refs #9556.

- `LegacyArchives` extracted from `Archives` into `legacy/archives.py`
- `Archives` in `manifest.py` is now 2.x only as all `self.legacy` branches removed
- `ArchivesInterface` added to `manifest.py` as a `@runtime_checkable` Protocol as mypy/pyright will catch drift between the two implementations
- `Manifest.__init__` picks `LegacyArchives` or `Archives` based on repository type via lazy imports

No logic changes. `borg transfer --from-borg1` keeps working through `borg.legacy.archives`. `borg serve` for v1 clients is unaffected — it only opens `LegacyRepository` for raw RPC byte-forwarding and never instantiates `LegacyArchives`.

Refs #9556

Checklist

- [x] PR is against master
- [x] New code has tests
- [x] Tests pass
- [x] Commit message references related issue

### CODE REVIEW

File: src/borg/legacy/archives.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
+"""Legacy archive list management for Borg 1.x repositories.
+
+In Borg 1.x the list of archives is embedded directly in the manifest blob
+as a dict {name: {"id": bytes, "time": str}}.  This class manages that dict.
+Used by ``borg transfer --from-borg1`` and ``borg serve`` (serving v1 clients).
```
Comment (ThomasWaldmann) [inline src/borg/legacy/archives.py:5]: I don't think `borg serve` uses `LegacyArchives` (or `Archives`).

Think of an encrypted repo - only the client has the key, the server can't even look into the blobs.

File: src/borg/manifest.py [hunk 2]
```diff
@@ -69,65 +70,92 @@ def get_first_and_last_archive_ts(archives_list):
...
 class Archives:
     """
-    Manage the list of archives.
+    Manage the list of archives for a Borg 2.x repository.
 
-    We still need to support the borg 1.x manifest-with-list-of-archives,
-    so borg transfer can work.
-    borg2 has separate items archives/* in the borgstore.
+    Each archive has a separate entry in borgstore at archives/<hex-id>.
+    The manifest blob itself carries an empty archives dict.
```
Comment (ThomasWaldmann) [inline src/borg/manifest.py:125]: Guess that is rather a leftover and should not be a documented feature. :-)

File: src/borg/manifest.py [hunk 3]
```diff
@@ -272,82 +286,55 @@ def _lookup_name(self, name, raw=False):
...
+                        name=archive_info["name"],
+                        id=archive_info["id"],
+                        ts=ts,
+                        tags=archive_info["tags"],
+                        user=archive_info["username"],
+                        host=archive_info["hostname"],
+                    )
+                return archive_info
+        return None  # id not in store, or archive metadata blob missing from repo
+        # TODO: add a test that calls get_by_id() with a non-existent id and asserts None is returned
```
Comment (ThomasWaldmann) [inline src/borg/manifest.py:313]: seen that?

File: src/borg/manifest.py [hunk 4]
```diff
@@ -272,82 +286,55 @@ def _lookup_name(self, name, raw=False):
...
 
     def create(self, name, id, ts, *, overwrite=False):
         assert isinstance(name, str)
         assert isinstance(id, bytes)
         if isinstance(ts, datetime):
             ts = ts.isoformat(timespec="microseconds")
         assert isinstance(ts, str)
-        if not self.legacy:
-            # we only create a directory entry, its name points to the archive item:
```
Comment (ThomasWaldmann) [inline src/borg/manifest.py:328]: looks like it killed my comment and replaced it by another.

**Comment by ThomasWaldmann** [comment]
This is a good first step for the archives list part of the Manifest.

In a further step (not this PR) we could try whether we still need other parts of the Manifest or whether we could get rid of them.

The main reason why this thing was called Manifest in borg 1.x (the manifest as a signed list of archives) is not in the manifest blob anymore in borg2.

File: src/borg/legacy/archives.py (hunk 1 above)
Comment (mr-raj12) [reply src/borg/legacy/archives.py:5]: thanks for correcting, removed it

File: src/borg/manifest.py (hunk 3 above)
Comment (mr-raj12) [reply src/borg/manifest.py:313]: added the test

File: src/borg/manifest.py (hunk 4 above)
Comment (mr-raj12) [reply src/borg/manifest.py:328]: restored your comment, apologies it missed my local review there

File: src/borg/manifest.py (hunk 2 above)
Comment (mr-raj12) [reply src/borg/manifest.py:125]: true
